datafusion/sql/src/select.rs

```diff
@@ -778,9 +777,12 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
         // Rewrite the HAVING expression to use the columns produced by the
         // aggregation.
         let having_expr_post_aggr = if let Some(having_expr) = having_expr_opt {
-            let having_expr_post_aggr =
-                rebase_expr(having_expr, &aggr_projection_exprs, input)?;
-
+            let having_expr_post_aggr = match having_expr {
```

Comment:
I wonder if this is common enough to add it as a function in `Expr` 

like 

```rust
impl Expr {
  /// return true if this expression is the boolean literal `true`
  pub fn is_true(&self) -> bool { ... }
  /// return true if this expression is the boolean literal `false`
  ...
  /// return true if this expression is the boolean literal `null`
  ...
}
```

(as a follow on PR of course 🤔 )

Reply:
I think it has nothing to do with it. The reason for this bug is `having_expr_post_aggr` wrongly transformed into a column instead of a boolean value.